File: archive/maximum-value-of-a-string-in-an-array.js
const maximumValue = function(strs) {
  let maxVal = 0
  strs.forEach(el => {
    if (isOnlyNumbers(el)) {
      maxVal = Math.max(Number(el), maxVal)
    } else {
      maxVal = Math.max(el.length, maxVal)
    }
  })

  return maxVal
}

function isOnlyNumbers (str) {
  let ans = true
  str.split('').forEach(el => {
    if (!(el.charCodeAt(0) >= 48 && el.charCodeAt(0) <=57)) {
      ans = false
    }
  })

  return ans
}

console.log(maximumValue(["alic3","bob","3","4","00000"]))

File: go-archive/maximum-value-of-a-string-in-an-array.go
func maximumValue(strs []string) int {
    ans:=0

    for _, v:=range strs{
        val, err:= strconv.Atoi(v)

        if (err == nil) {
            if (val > ans) {ans = val}
        } else {
            if (len(v) > ans) {ans = len(v)}
        }
    }

    return ans
}